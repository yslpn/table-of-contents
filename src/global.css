*,
*::before,
*::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

html,
body {
  font-family:
    system-ui,
    -apple-system,
    'Segoe UI',
    Roboto,
    'Helvetica Neue',
    'Noto Sans',
    'Liberation Sans',
    Arial,
    sans-serif,
    'Apple Color Emoji',
    'Segoe UI Emoji',
    'Segoe UI Symbol',
    'Noto Color Emoji';
  color: #fff;
  background-color: #19191c;

  @media (prefers-color-scheme: light) {
    color: #213547;
    background-color: #fff;
  }
}
