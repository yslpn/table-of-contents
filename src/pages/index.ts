export { HomePage } from './home/ui/HomePage';
