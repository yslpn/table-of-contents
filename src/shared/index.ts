export { Spinner } from './ui/spinner/Spinner';
export { MainLayout } from './ui/main-layout/MainLayout';

export { fetcher } from './api/fetcher';

export type { IPage, IEntities, ITableOfContentsData } from './types/toc-data';
