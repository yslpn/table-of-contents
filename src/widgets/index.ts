export { Header } from './header/ui/Header';
export { TableOfContents } from './table-of-contents/ui/table-of-contents/TableOfContents';
