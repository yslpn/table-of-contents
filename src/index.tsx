import { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';

import './global.css';

import { HomePage } from './pages';

const rootElement = document.getElementById('root');

if (rootElement instanceof HTMLElement) {
  ReactDOM.createRoot(rootElement).render(
    <StrictMode>
      <HomePage />
    </StrictMode>,
  );
} else {
  console.error("Could not find the 'root' element.");
}
