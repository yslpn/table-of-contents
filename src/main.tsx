import { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';

import { HomePage } from './pages';

import './global.css';

const rootElement = document.getElementById('root');

if (rootElement instanceof HTMLElement) {
  ReactDOM.createRoot(rootElement).render(
    <StrictMode>
      <HomePage />
    </StrictMode>,
  );
} else {
  console.error("Could not find the 'root' element.");
}
