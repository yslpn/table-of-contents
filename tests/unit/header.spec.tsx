import renderer from 'react-test-renderer';
import { Header } from '../../src/widgets';

test('Header component should match the snapshot', () => {
  const component = renderer.create(<Header />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
