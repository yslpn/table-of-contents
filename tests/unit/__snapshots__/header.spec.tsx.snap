// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match the snapshot 1`] = `
<header
  className="header"
  data-test-id="page-header"
>
  <div
    className="container"
  >
    <p
      className="text"
    >
      <IconComponent />
      Product name
    </p>
  </div>
</header>
`;
